### PR TITLE
Fix build warnings when heavy checking happens

### PR DESCRIPTION
**Pull request by ngaro**
- Fix dead link in documentation
- Rewrite other link to work around unicode problems
- Use unsigned int for a size that can't be negative

More fixes are coming...

### CODE REVIEW

File: doc/docs.xml [hunk 1]
```diff
@@ -288,17 +288,13 @@
 	<refsect1>
 		<title>See Also</title>
 		<para>
-			<ulink url="http://conky.sourceforge.net/">
+			<ulink url="http://conky.sourceforge.net">
 				http://conky.sourceforge.net/</ulink>
```
Comment (lasers) [inline doc/docs.xml]: Can you change this to `https://github.com/brndnmtthws/conky` (x2)? It redirects. Skip that.

File: doc/docs.xml [hunk 2]
```diff
@@ -288,17 +288,13 @@
 	<refsect1>
 		<title>See Also</title>
 		<para>
-			<ulink url="http://conky.sourceforge.net/">
+			<ulink url="http://conky.sourceforge.net">
```
Comment (lasers) [inline doc/docs.xml]: Can you change this to `https://github.com/brndnmtthws/conky` (x2)? It redirects. Skip that.

File: doc/docs.xml [hunk 3]
```diff
@@ -288,17 +288,13 @@
...
 				http://conky.sourceforge.net/</ulink>
 		</para>
 		<para>
-			<ulink url="http://www.sourceforge.net/projects/conky">
+			<ulink url="http://sourceforge.net/projects/conky">
```
Comment (lasers) [inline doc/docs.xml]: :lock: `https://sourceforge.net/projects/conky`

File: src/tailhead.cc [hunk 4]
```diff
@@ -175,7 +175,7 @@ static void print_tailhead(const char *type, struct text_object *obj, char *p,
               endofstring = strlen(p);
             }
           } else if (strcmp(type, "tail") == 0) {
-            fseek(fp, -p_max_size, SEEK_END);
+            fseek(fp, (long) -p_max_size, SEEK_END);
```
[the remark this answers is not in the document]
Comment (ngaro) [reply src/tailhead.cc]: Don't approve my request yet. I just noticed that my changes broke regular building.
Help is on the way...

File: doc/docs.xml (hunk 3 above)
Comment (lasers) [reply doc/docs.xml]: Code :nose:: The URLs are not identical.

File: src/tailhead.cc (hunk 4 above)
[the remark this answers is not in the document]
Comment (ngaro) [reply src/tailhead.cc]: Fixed forgotten parts of commit, building works fine again

File: doc/docs.xml [hunk 5]
```diff
@@ -288,17 +288,13 @@
 	<refsect1>
 		<title>See Also</title>
 		<para>
-			<ulink url="http://conky.sourceforge.net/">
+			<ulink url="https://github.com/brndnmtthws/conky">
 				http://conky.sourceforge.net/</ulink>
```
Comment (brndnmtthws) [inline doc/docs.xml]: Should should probably be `https://github.com/brndnmtthws/conky` since you're in there anyway.

File: doc/docs.xml [hunk 6]
```diff
@@ -288,17 +288,13 @@
...
 		</para>
 		<para>
-			<ulink url="http://www.sourceforge.net/projects/conky">
+			<ulink url="http://sourceforge.net/projects/conky">
 				http://www.sourceforge.net/projects/conky</ulink>
```
Comment (brndnmtthws) [inline doc/docs.xml]: Ditto here.

**Comment by ngaro** [comment]
In maintainer-mode "make docs" only builds without warning if I strip the 'y' at the end of the url...
Otherwise it complains with:
<standard input>:3573: warning: can't find special character `u3008'
<standard input>:3573: warning: can't find special character `u3009'

I'll assume we'll ignore this for now ?

**Comment by ngaro** [comment]
I don't get that last sonarcloud issue, similar thing happens all over the place. Why is this the only complaint ?

**Comment by brndnmtthws** [comment]
@ngaro sonar only runs on lines of code that are changed. Thus, all the existing code isn't affected. In this case, the C macro should be unwrapped.

I went ahead and fixed it.

**Comment by ngaro** [comment]
Ok, All problems found with -Wall are solved here.
I'll leave the 2 sonarcloud issues up to you.
Every thing should still work in theory, but that's only theory... Don't give this code a version number ;-)

**Comment by ngaro** [comment]
My attempt to work around those 2 similar sonarcloud issues didn't work.
I'll leave this to someone else, meanwhile this can already be merged

**Comment by brndnmtthws** [comment]
I merged #592 in place of this.

**Pull request closed without merging**